### PR TITLE
Backport #504, #496, #499 (Adapter gone detection)

### PR DESCRIPTION
**Pull request by gytsto**
### Problem
When adapter is dead, we do nothing and proceed as usual.

### Solution
Backport of https://github.com/NordSecurity/libtelio/pull/504 ; https://github.com/NordSecurity/libtelio/pull/496 ; https://github.com/NordSecurity/libtelio/pull/499

### :ballot_box_with_check: Definition of Done checklist
- [x] Commit history is clean ([requirements](../blob/main/docs/git_commit_messages_requirements.md))
- [x] README.md is updated
- [x] changelog.md is updated
- [x] Functionality is covered by unit or integration tests

### CODE REVIEW

**Review by matislovas — APPROVED** [review]

**Review by matislovas — APPROVED** [review]

**Review by jjanowsk — COMMENTED** [review]
+2 for .github directory

File: changelog.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
+### v4.2.9
```
Comment (packgron) [inline changelog.md:1]: @matislovas I think it would be smart to wait for @tomaszklak changes, and release together with it

**Review by packgron — APPROVED** [review]
+0.5

File: changelog.md (hunk 1 above)
Comment (gytsto) [reply changelog.md:1]: I think he wanted to do release later anyway, I just bumped changelog version in advance, so another PR wouldn't be needed if it was decided to release now

Comment (gytsto) [reply changelog.md:1]: But now I see that Cargo.toml wasn't changed anyway :smile:

**Pull request merged**